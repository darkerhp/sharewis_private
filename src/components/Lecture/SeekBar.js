import React from 'react';
import ReactNative from 'react-native';

import Duration from '../Duration';

const { PropTypes } = React;
const { Platform, View, StyleSheet, Slider, Dimensions } = ReactNative;

const DEVICE_WIDTH = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    flexDirection: 'row',
  },
  slider: {
    width: (DEVICE_WIDTH - 90) * 0.8,
    ...Platform.select({
      android: {
        backgroundColor: '#eee',
      },
    }),
  },
  timeTextContainer: {
    flex: 1,
    justifyContent: 'center',
    width: 60,
  },
  timeText: {
    color: '#bdbdbd',
    fontSize: 16,
    fontWeight: 'bold',
    opacity: 0.8,
    padding: 0,
    textAlign: 'center',
  },
});

const SeekBar = ({ currentTime, estimatedTime, onValueChange }) => {
  const timeEnd = estimatedTime - currentTime;
  return (
    <View style={styles.container}>
      <Duration
        estimatedTime={currentTime}
        format={'mm:ss'}
        containerStyleId={styles.timeTextContainer}
        durationStyleId={styles.timeText}
      />
      <Slider
        maximumValue={estimatedTime}
        value={currentTime}
        onValueChange={value => onValueChange(value)}
        style={styles.slider}
      />
      <Duration
        estimatedTime={timeEnd < 0 ? 0 : timeEnd}
        format={'mm:ss'}
        containerStyleId={styles.timeTextContainer}
        durationStyleId={styles.timeText}
        prefixText={'-'}
      />
    </View>
  );
};


SeekBar.propTypes = {
  currentTime: PropTypes.number.isRequired,
  estimatedTime: PropTypes.number.isRequired,
  onValueChange: PropTypes.func.isRequired,
};

export default SeekBar;
