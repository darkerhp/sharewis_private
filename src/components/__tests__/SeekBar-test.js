/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import SeekBar from '../Lecture/SeekBar';

jest.mock('../Duration', () => 'Duration');
jest.mock('Slider', () => 'Slider');

describe('SeekBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SeekBar
        currentTime={10}
        estimatedTime={100}
        onValueChange={() => null}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
