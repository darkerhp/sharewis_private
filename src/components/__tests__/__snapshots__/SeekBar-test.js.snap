exports[`SeekBar renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around"
    }
  }>
  <Duration
    containerStyleId={
      Object {
        "justifyContent": "center",
        "width": 45
      }
    }
    durationStyle={
      Object {
        "backgroundColor": "white",
        "fontSize": 13,
        "fontWeight": "normal",
        "opacity": 0.6,
        "padding": 0,
        "textAlign": "center"
      }
    }
    estimatedTime={10}
    format="mm:ss" />
  <Slider
    maximumValue={100}
    onValueChange={[Function onValueChange]}
    style={
      Object {
        "width": 256.5
      }
    }
    value={10} />
  <Duration
    containerStyleId={
      Object {
        "justifyContent": "center",
        "width": 45
      }
    }
    durationStyle={
      Object {
        "backgroundColor": "white",
        "fontSize": 13,
        "fontWeight": "normal",
        "opacity": 0.6,
        "padding": 0,
        "textAlign": "center"
      }
    }
    estimatedTime={90}
    format="mm:ss"
    prefixText="-" />
</View>
`;
